src/mbgl/map/sprite.cpp

```diff
@@ -43,8 +44,8 @@ Sprite::Sprite(const std::string& baseUrl, float pixelRatio_)
         return;
     }
 
-    std::string spriteURL(baseUrl + (pixelRatio_ > 1 ? "@2x" : "") + ".png");
-    std::string jsonURL(baseUrl + (pixelRatio_ > 1 ? "@2x" : "") + ".json");
+    std::string spriteURL = mbgl::util::mapbox::normalizeSpriteURL(baseUrl, (pixelRatio_ > 1 ? "@2x" : ""), ".png", "NOCOMMIT");
+    std::string jsonURL = mbgl::util::mapbox::normalizeSpriteURL(baseUrl, (pixelRatio_ > 1 ? "@2x" : ""), ".json", "NOCOMMIT");
```

Comment:
We've made Mapbox URL normalization the responsibility of `FileSource` implementations, to permit the implementation in api-gl to load resources directly from the DB and not use an access token at all. We might need some changes to the `FileSource` interface to accommodate `mapbox://` sprite URLs. Mainly I think the implementation needs to be able distinguish sprite JSON requests from other JSON requests (if there are any).